packages/gatsby-theme-docs/src/components/markdown-fragment-image.js

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+import { useResizedImagesByPath } from '../hooks/use-resized-images';
+
+const MarkdownFragmentImage = (props) => {
+  const image = useResizedImagesByPath(props.src);
+  if (!image) {
+    return null;
+  }
+
+  // NOTE: the HTML structure below is taken from the `gatsby-remark-images` plugin,
+  // as we want to replicate the same behaviour and structure.
+  const ratio = `${(1 / image.aspectRatio) * 100}%`;
```

Comment:
This is taken from here: https://github.com/gatsbyjs/gatsby/blob/416dca7f67894dc34a48df4435b587ec3f2ac251/packages/gatsby-remark-images/src/index.js#L281